File: api/org.integratedmodelling.klab.api/src/org/integratedmodelling/klab/rest/AuthenticationResponse.java
package org.integratedmodelling.klab.rest;

public abstract class AuthenticationResponse {
	
	private AuthenticatedIdentity userData;
	private HubReference hubReference;
	
	public AuthenticatedIdentity getUserData() {
		return userData;
	}
	public void setUserData(AuthenticatedIdentity userData) {
		this.userData = userData;
	}
	public HubReference getHubReference() {
		return hubReference;
	}
	public void setHubReference(HubReference hubReference) {
		this.hubReference = hubReference;
	}

}

File: klab.core/src/main/java/org/integratedmodelling/klab/data/rest/resources/responses/AuthenticationResponse.kt
package org.integratedmodelling.klab.data.rest.resources.responses

import org.integratedmodelling.klab.data.rest.resources.AuthenticatedIdentity
import org.integratedmodelling.klab.data.rest.resources.NodeReference

/**
 * Auth response.
 */
data class AuthenticationResponse (
		val userData: AuthenticatedIdentity,
		val nodes: List<NodeReference>,
		val authenticatingNodeId: String) {
	constructor() : this(AuthenticatedIdentity(), ArrayList(), "")
}